lib/LaTeXML/Package.pm

```diff
@@ -974,6 +975,32 @@ sub TokenizeInternal {
   local $STATE = $STY_CATTABLE;
   return LaTeXML::Core::Mouth->new($string)->readTokens; }
 
+# Check whether all these things are "empty" (spaces are empty!!)
+# short-circuit: return 0 quickly if anything is NOT empty.
+# Will we need to distinguish between "empty" and "not visible"??
+sub IsEmpty {
+  my (@things) = @_;
+  foreach my $thing (@things) {
+    my $ref = ref $thing;
+    if    (!$thing)                          { }
+    elsif ($ref eq 'LaTeXML::Core::Comment') { }
+    elsif ($ref eq 'LaTeXML::Core::Tokens') {
+      return 0 unless IsEmpty($thing->unlist); }
+    elsif ($ref eq 'LaTeXML::Core::Token') {
+      my $cc = $$thing[1];
+      return 0 if ($cc == CC_LETTER) || ($cc == CC_OTHER) || ($cc == CC_ACTIVE) || ($cc == CC_CS); }
```

Comment:
Hm, this seems incomplete? A T_BEGIN, T_END, T_MATH, T_ALIGN, T_PARAM aren't empty either. Maybe it's somewhat briefer to invert the condition:
```perl
return 0 unless ($cc == CC_COMMENT) || ($cc == CC_MARKER) || ($cc == CC_IGNORE);
```

Reply:
though if this is meant to be used for a single Token, I'd also wonder if we shouldn't treat all tokens that have empty text as empty as well.
```perl
return 0 unless ($cc == CC_COMMENT) || ($cc == CC_MARKER) || ($cc == CC_IGNORE) 
  || (!length($$thing[0]));
```

Reply:
I think you have the logic flipped; this line should short-circuit immediately returning "false" if we encounter a token that  is *not* empty.
OTOH, your point is good: should we consider `T_LETTER("")` to be empty?
OTOOH, I expect this issue may have to be revisited to distinguish between truly empty, and just spacey things? Hopefully I can defer that concern?